dmoj/contrib/coci.py

```diff
@@ -0,0 +1,26 @@
+import re
+
+from dmoj.contrib.testlib import ContribModule as TestlibContribModule
+from dmoj.error import InternalError
+from dmoj.result import CheckerResult
+
+
+class ContribModule(TestlibContribModule):
+    name = 'coci'
+    repartial = re.compile(br'^partial ((\d+)\/(\d*[1-9]\d*))\n$')
+
+    @classmethod
+    def parse_return_code(cls, proc, executor, point_value, time_limit, memory_limit, feedback, name, stderr):
+        if proc.returncode == cls.PARTIAL:
+            match = cls.repartial.match(stderr)

```

Comment:
Actually, from lessons learned in https://github.com/DMOJ/judge-server/pull/708, this should be `search`.